### ISSUE TITLE
[Bug] Pad with mode="wrap" causes compiler to hang (infinite loop)

### ISSUE THREAD

**Issue by wuyii8941** [iree-org/iree#24585]
### What happened?

The `Pad` op with `mode="wrap"` causes `iree-compile` to hang indefinitely (tested >120 seconds on a trivial 4×4 input). The compiler appears to enter an infinite loop. Other pad modes (`constant`, `reflect`, `edge`) compile and run correctly on the same input.

ONNX Runtime handles wrap padding correctly.

This also affects PyTorch models that use `torch.nn.functional.pad(mode='circular')`, which exports as `Pad(mode='wrap')` in ONNX.

### Other pad modes (control, same input and pads)

| Mode | Result |
|---|---|
| `constant` | Compiles and runs correctly |
| `reflect` | Compiles and runs correctly |
| `edge` | Compiles and runs correctly |
| `wrap` | **Hangs indefinitely** |

### Impact

Wrap padding is used for circular/periodic padding in signal processing and models that require toroidal boundary conditions. In PyTorch, `torch.nn.functional.pad(mode='circular')` exports as ONNX `Pad(mode='wrap')`, so any PyTorch model using circular padding will trigger this hang.

### Expected behavior

Compilation should either succeed and produce correct output, or fail with a clear error message — not hang indefinitely.

### Environment

- IREE compiler: `3.12.0rc20260515`
- Target: `llvm-cpu` (generic)
- ONNX opset: 19

### Steps to reproduce

```python
import numpy as np
import onnx
from onnx import helper, TensorProto, numpy_helper

X = helper.make_tensor_value_info("X", TensorProto.FLOAT, [1, 1, 4, 4])
Y = helper.make_tensor_value_info("Y", TensorProto.FLOAT, None)
pads = numpy_helper.from_array(np.array([0,0,1,1,0,0,1,1], dtype=np.int64), "pads")
nodes = [helper.make_node("Pad", ["X", "pads"], ["Y"], mode="wrap")]
graph = helper.make_graph(nodes, "test", [X], [Y], initializer=[pads])
model = helper.make_model(graph, opset_imports=[helper.make_opsetid("", 17)])
onnx.save(model, "/tmp/pad_wrap.onnx")
```

```bash
iree-import-onnx /tmp/pad_wrap.onnx -o /tmp/pad_wrap.mlir  # OK
iree-compile --iree-hal-target-backends=llvm-cpu --iree-llvmcpu-target-cpu=generic \
  /tmp/pad_wrap.mlir -o /tmp/pad_wrap.vmfb
# Hangs indefinitely — no output, no error, no progress
```

**Comment by devtbi** [comment]
root-caused with claude:

- `AtenPadOp` is registered **illegal** in `DecomposeComplexOps` (`addPatternIfTargetOpIsIllegal<DecomposeAtenPadOp>`, `DecomposeComplexOps.cpp:13573`) — it must be decomposed away.
- `DecomposeAtenPadOp` (`DecomposeComplexOps.cpp:9013`) only handles `constant` / `reflect` / `replicate`; for `circular` it falls through to `notifyMatchFailure(op, "unsupported mode: circular")` at `:9116`, so the op is **never legalized**.
- The pass runs the greedy driver with **no iteration cap**: `config.setMaxIterations(GreedyRewriteConfig::kNoLimit)` (`DecomposeComplexOps.cpp:13713`). With a permanently-illegal op the worklist never reaches fixpoint → **infinite spin in `torch-decompose-complex-ops`**

reproducer
`iree-compile --iree-hal-target-device=local  --iree-hal-local-target-device-backends=llvm-cpu pad_circular_repro.mlir`
```
// reproducer — torch.aten.pad with mode="circular".
//
// This is what convert-torch-onnx-to-torch emits for ONNX
// Pad mode="wrap" (DefaultDomainGtoP.cpp:3039 maps wrap->circular)
//   x: 1x1x4x4, pads [0,0,1,1, 0,0,1,1] -> 1x1x6x6.
func.func @pad_circular(%arg0: !torch.vtensor<[1,1,4,4],f32>) -> !torch.vtensor<[1,1,6,6],f32> {
  %int0 = torch.constant.int 0
  %int1 = torch.constant.int 1
  // torch.aten.pad pad list is innermost-dim-first: [W_begin, W_end, H_begin, H_end, ...]
  %pads = torch.prim.ListConstruct %int1, %int1, %int1, %int1, %int0, %int0, %int0, %int0 : (!torch.int, !torch.int, !torch.int, !torch.int, !torch.int, !torch.int, !torch.int, !torch.int) -> !torch.list<int>
  %mode = torch.constant.str "circular"
  %none = torch.constant.none
  %0 = torch.aten.pad %arg0, %pads, %mode, %none : !torch.vtensor<[1,1,4,4],f32>, !torch.list<int>, !torch.str, !torch.none -> !torch.vtensor<[1,1,6,6],f32>
  return %0 : !torch.vtensor<[1,1,6,6],f32>
}
```
